### PR TITLE
Add release workflow

### PR DESCRIPTION
**Pull request by squaremo**
This is copied in whole from source-controller, and adapted by replacing `source-controller` with `image-reflector-controller` (only).

### CODE REVIEW

**Comment by squaremo** [comment]
(The only way to test this is to tag the repo)

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        uses: actions/create-release@latest
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          tag_name: ${{ github.ref }}
+          release_name: ${{ github.ref }}
+          draft: false
+          prerelease: true
+          body: |
+            [CHANGELOG](https://github.com/fluxcd/image-reflector-controller/blob/master/CHANGELOG.md)
```
Comment (hiddeco) [inline .github/workflows/release.yml:105]:
```suggestion
            [CHANGELOG](https://github.com/fluxcd/image-reflector-controller/blob/main/CHANGELOG.md)
```

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    tags:
+      - 'v*'
+
+jobs:
+  build-push:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup Kustomize
+        uses: fluxcd/pkg/actions/kustomize@master
```
Comment (hiddeco) [inline .github/workflows/release.yml:13]:
```suggestion
        uses: fluxcd/pkg/actions/kustomize@main
```

File: .github/workflows/release.yml (hunk 1 above)
Comment (squaremo) [reply .github/workflows/release.yml:105]: Good spot

**Review by hiddeco — APPROVED** [review]

**Pull request merged**